cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/web/SearchController.java

```diff
@@ -0,0 +1,63 @@
+package edu.asu.diging.cord19.explorer.web;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.PostConstruct;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import edu.asu.diging.cord19.explorer.core.mongo.impl.SearchProviderRegistry;
+import edu.asu.diging.cord19.explorer.web.model.SearchType;
+
+@Controller
+public class SearchController {
+    
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Autowired
+    private SearchProviderRegistry searchProviderRegistry;
+    
+    private Map<SearchType, String> views = new HashMap<>();
+
+    @RequestMapping(value = "/{searchType}/search")
+    public String search(@PathVariable("searchType") String searchType, @RequestParam("search") String query,
+            Model model, @PageableDefault(size = 20) Pageable pageable, RedirectAttributes redirectAttrs) {
+
+        try {
+            searchType = searchType.toUpperCase();
+            long count = searchProviderRegistry.getProvider(SearchType.valueOf(searchType)).getTotalResults(query);
+            model.addAttribute("matchedResultsPage", searchProviderRegistry.getProvider(SearchType.valueOf(searchType))
+                    .search(query, (long) pageable.getPageNumber(), pageable.getPageSize()));
+            model.addAttribute("total", count);
+            model.addAttribute("query", query);
+            model.addAttribute("page", pageable.getPageNumber());
+            model.addAttribute("pageCount",
+                    count / pageable.getPageSize() + (count % pageable.getPageSize() > 0 ? 1 : 0));
+            return views.get(SearchType.valueOf(searchType));
+        } catch (IllegalArgumentException e) {
```

Comment:
where is the exception thrown? the try/catch block should be around that piece of code, not more.